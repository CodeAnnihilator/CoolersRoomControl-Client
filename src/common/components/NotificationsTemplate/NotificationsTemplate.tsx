import React, {Component} from 'react';
import {View, ScrollView, Text} from 'react-native';
import dayjs from 'dayjs';

import NotificationCard from '../NotificationCard/NotificationCard';
import ServicesNotificationCard from '../ServicesNotificationCard/ServicesNotificationCard';

import styles from './notificationsTemplateStyles';

interface INotification {
	id: number;
	type: string;
	date: string;
	time: string;
	operation?: string;
	temperature?: string;
	humidity?: string;
	weekly?: boolean;
	description?: string;
	message?: string;
	model?: string;
	serial?: string;
	roomId: number;
}

interface IProps {
	notifications: any;
	dates: number[];
}

class Notifications extends Component<IProps> {
	public renderContent = (notification: INotification) => {
		switch (notification.type) {
			case 'notification': {
				return <NotificationCard notification={notification} key={notification.id} />;
			}
			case 'servicesNotification': {
				return <ServicesNotificationCard notification={notification} key={notification.id} />;
			}
		}
	}

	public render() {
		const {notifications, dates} = this.props;

		return (
			<ScrollView style={styles.container}>
				{dates.length
					? dates.map((el: number, index: number) => (
						<View key={el} style={styles.cardWrapper}>
							<View style={styles.border}>
								<Text style={styles.date}>{dayjs(el).format('DD MMM')}</Text>
							</View>
							<View style={styles.content}>
								{notifications.length && notifications[index].map(this.renderContent)}
							</View>
						</View>
					))
					: (
						<View style={styles.notFound}>
							<Text style={styles.notFoundText}>
								Notifications not found
							</Text>
						</View>
					)
				}
			</ScrollView>
		);
	}
}

export default Notifications;
