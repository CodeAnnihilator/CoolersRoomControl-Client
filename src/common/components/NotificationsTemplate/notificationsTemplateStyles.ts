import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	container: {
		flex: 1,
	},
	cardWrapper: {
		flex: 1,
		flexDirection: 'row',
	},
	border: {
		display: 'flex',
		alignItems: 'center',
		width: 74,
		backgroundColor: '$gray100',
	},
		date: {
		fontSize: 14,
		fontWeight: 'bold',
		paddingTop: 36,
		color: '$black',
		textTransform: 'uppercase',
	},
	content: {
		flex: 1,
		flexDirection: 'column',
	},
});