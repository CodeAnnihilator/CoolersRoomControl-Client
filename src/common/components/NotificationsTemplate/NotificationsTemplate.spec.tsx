import * as React from 'react';
import renderer from 'react-test-renderer';

import NotificationsTemplate from './NotificationsTemplate';

describe('NotificationsTemplate', () => {
	it('+++capturing Snapshot of NotificationsTemplate', () => {
		const dates = [1556527572377, 1556527583146];
		const notifications = [
			[{
				id: 1,
				type: 'type',
				date: 'date',
				time: 'time',
				operation: 'operation',
				temperature: 'temperature',
				humidity: 'humidity',
				weekly: true,
				description: 'description',
				message: 'message',
				model: 'model',
				serial: 'serial',
				roomId: 1,
			}],
			[{
				id: 1,
				type: 'type',
				date: 'date',
				time: 'time',
				operation: 'operation',
				temperature: 'temperature',
				humidity: 'humidity',
				weekly: true,
				description: 'description',
				message: 'message',
				model: 'model',
				serial: 'serial',
				roomId: 1,
			}],
		];
		const renderedValue =  renderer.create(
			<NotificationsTemplate dates={dates} notifications={notifications} />,
		).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
