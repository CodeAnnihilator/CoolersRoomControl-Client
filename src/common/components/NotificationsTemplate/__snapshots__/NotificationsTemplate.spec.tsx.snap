// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsTemplate +++capturing Snapshot of NotificationsTemplate 1`] = `null`;
